src/azure-cli/azure/cli/command_modules/acr/_utils.py

```diff
@@ -431,7 +438,36 @@ def parse_actions_from_repositories(allow_or_remove_repository):
 
     return actions
 
+def prepare_source_location(source_location, client_registries, registry_name, resource_group_name):
+    if source_location.lower() == NULL_CONTEXT:
+        source_location = None
+    elif os.path.exists(source_location):
+        if not os.path.isdir(source_location):
+            raise CLIError(
+                "Source location should be a local directory path or remote URL.")
+
+        tar_file_path = os.path.join(tempfile.gettempdir(
+        ), 'task_run_archive_{}.tar.gz'.format(uuid.uuid4().hex))
```

Comment:
can we have a dynamic name here? 
I would make easier to track the source based on the command (e.g. build_archive, run_archive, task_run_archive) when troubleshooting.